src/pages/AddressInfoPage/index.tsx

```diff
@@ -0,0 +1,321 @@
+import Timestamp from '@/components/Timestamp'
+import { useGlobalContext } from '@/contexts/global'
+import usePageNumber from '@/hooks/usePageNumber'
+import ExportAddressTXsModal from '@/modals/ExportAddressTXsModal'
+import { deviceBreakPoints } from '@/styles/globalStyles'
+import { AddressDataResult, AddressTransactionsResult } from '@/types/addresses'
+import { Asset } from '@/types/assets'
+import { ALPH } from '@/utils/assets'
+import { formatNumberForDisplay } from '@/utils/strings'
+
+import AddressTransactionRow from './AddressTransactionRow'
+import AssetList from './AssetList'
+import AddressInfoGrid from './InfoGrid'
+
+type ParamTypes = {
+  id: string
+}
+
+const TransactionInfoPage = () => {
+  const theme = useTheme()
+  const { id } = useParams<ParamTypes>()
+  const { client, setSnackbarMessage, networkType } = useGlobalContext()
+
+  const [addressData, setAddressData] = useState<AddressDataResult>()
+  const [addressTransactions, setAddressTransactions] = useState<AddressTransactionsResult>()
+
+  const [addressDataLoading, setAddressDataLoading] = useState(true)
+  const [txLoading, setTxLoading] = useState(true)
+
+  const [addressWorth, setAddressWorth] = useState<number | undefined>(undefined)
+
+  const [exportModalShown, setExportModalShown] = useState(false)
+
+  // Default page
+  const pageNumber = usePageNumber()
+
+  useEffect(() => {
+    if (!client || !id) return
+
+    const fetch = async () => {
+      setAddressDataLoading(true)
+      setAddressData(undefined)
+      try {
+        const addressData = await fetchAddressData(client, id)
+        setAddressData(addressData)
+      } catch (error) {
+        console.error(error)
+        setSnackbarMessage({
+          text: getHumanReadableError(error, 'Error while fetching address data'),
+          type: 'alert'
+        })
+      } finally {
+        setAddressDataLoading(false)
+      }
+    }
+
+    fetch()
+  }, [client, id, setSnackbarMessage])
+
+  useEffect(() => {
+    if (!client || !id) return
+
+    const fetch = async () => {
+      setTxLoading(true)
+      setAddressTransactions(undefined)
+      try {
+        const transactionsData = await fetchAddressTransactions(client, id, pageNumber)
+        setAddressTransactions(transactionsData)
+      } catch (error) {
+        console.error(error)
+        setSnackbarMessage({
+          text: getHumanReadableError(error, "Error while fetching address' transactions"),
+          type: 'alert'
+        })
+      } finally {
+        setTxLoading(false)
+      }
+    }
+
+    fetch()
+  }, [client, id, pageNumber, setSnackbarMessage])
+
+  // Asset price (appox).
+  // TODO: when listed tokens, add resp. prices. ALPH only for now.
+  useEffect(() => {
+    setAddressWorth(undefined)
+
+    const getAddressWorth = async () => {
+      const balance = addressData?.details.balance
+      if (!balance) return
+
+      const price = await fetchAssetPrice('alephium')
```

Comment:
I guess it's here where we could use a try/catch to avoid network errors, similar to how we catch them above in this file

Reply:
Didn't see your comment before, it's what I've done 💯